src/components/hosts/AdditionalNTPSourcesDialog.tsx

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { Formik, FormikHelpers } from 'formik';
+import * as Yup from 'yup';
+import {
+  Alert,
+  Form,
+  Modal,
+  ModalVariant,
+  ModalBoxBody,
+  ModalBoxFooter,
+  Button,
+  AlertActionCloseButton,
+  AlertVariant,
+  AlertActionLink,
+} from '@patternfly/react-core';
+import { Cluster, ClusterUpdateParams } from '../../api/types';
+import { ntpSourceValidationSchema } from '../ui/formik/validationSchemas';
+import TextAreaField from '../ui/formik/TextAreaField';
+import { patchCluster } from '../../api/clusters';
+import { updateCluster } from '../../features/clusters/currentClusterSlice';
+import { getErrorMessage, handleApiError } from '../../api/utils';
+import { trimCommaSeparatedList } from '../ui/formik/utils';
+
+type AdditionalNTPSourcesDialogToggleProps = {
+  cluster: Cluster;
+  onToggle?: (isOpen: boolean) => void;
+};
+
+export const AdditionalNTPSourcesDialogToggle: React.FC<AdditionalNTPSourcesDialogToggleProps> = ({
+  cluster,
+  onToggle,
+}) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const setOpen = (isOpen: boolean) => {
+    onToggle && onToggle(isOpen);
+    setIsOpen(isOpen);
+  };
+
+  return (
+    <>
+      <AlertActionLink onClick={() => setOpen(true)}>Add NTP sources</AlertActionLink>
+      <AdditionalNTPSourcesDialog
+        isOpen={isOpen}
+        cluster={cluster}
+        onClose={() => setOpen(false)}
+      />
+    </>
+  );
+};
+
+type AdditionalNTPSourcesDialogProps = {
+  isOpen: boolean;
+  cluster: Cluster;
+  onClose: () => void;
+};
+
+const AdditionalNTPSourcesDialog: React.FC<AdditionalNTPSourcesDialogProps> = ({
+  cluster,
+  isOpen,
+  onClose,
+}) => {
+  return (
+    <Modal
+      aria-label="Configure additional NTP sources"
+      title="Configure additional NTP sources"
+      isOpen={isOpen}
+      onClose={onClose}
+      variant={ModalVariant.small}
+      hasNoBodyWrapper
+    >
+      <AdditionalNTPSourcesForm cluster={cluster} onClose={onClose} />
+    </Modal>
+  );
+};
+
+export default AdditionalNTPSourcesDialog;
```

Comment:
For better codestyle, this should be at the end of the file

Reply:
done